### ISSUE TITLE
ByRef variable passed with parentheses should pass a copy

### ISSUE THREAD

**Issue by Kr00l** [WaynePhillipsEA/twinbasic#416]
**Describe the bug**
A string variable which is passed to a ByRef method with parentheses should be passed as a copy.

**To Reproduce**
```
Sub Main()
    
    Dim Str As String
    Str = "Hello"
    AppendStr (Str)
    Debug.Print Str ' should print "Hello" as Str is evaluated (parentheses) and a copy should be passed.
    AppendStr Str
    Debug.Print Str ' should print "Hello World". However, tB prints "Hello World World"
    
End Sub

Sub AppendStr(ByRef Str As String)
    Str = Str + " World"
End Sub
```

**Expected behavior**
tB should print "Hello" and "Hello World", but instead it prints "Hello World" and "Hello World World".

**Screenshots**
n/a

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10]
 - twinBASIC compiler version [e.g. 0.10.5711]

**Additional context**
This is currently a behavior mismatch between tB and VBx.

**Comment by Kr00l** [comment]
Here is a msdn article of the language feature to force a parameter as ByVal with parentheses.
[How to: Force an Argument to Be Passed by Value (Visual Basic)](https://docs.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/procedures/how-to-force-an-argument-to-be-passed-by-value)

**Comment by WaynePhillipsEA** [comment]
Thanks @Kr00l.  Indeed, the bracketed expressions are being reduced/simplified incorrectly here. We already account for parenthesis handling around objects (to evaluate the default members), so this should be an easy fix tomorrow.

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.5725, thanks
